pandas/_libs/tslib.pyx

```diff
@@ -1233,7 +1233,10 @@ cdef class _Timestamp(datetime):
         return datetime.__sub__(self, other)
 
     cpdef _get_field(self, field):
-        out = get_date_field(np.array([self.value], dtype=np.int64), field)
+        val = self.value
+        if self.tz is not None and not _is_utc(self.tz):
+            val = tz_convert_single(self.value, 'UTC', self.tz)
+        out = get_date_field(np.array([val], dtype=np.int64), field)
         return int(out[0])
 
     cpdef _get_start_end_field(self, field):
```

Comment:
Not sure what you mean with `get_date_field`, but this example seems to be wrong:

```
In [3]: pd.Timestamp("2014-01-01 00:00:00+01:00").is_month_start
Out[3]: False

In [4]: pd.Timestamp("2014-01-01 00:00:00").is_month_start
Out[4]: True
```